sql/src/main/java/io/crate/executor/transport/SnapshotRestoreDDLDispatcher.java

```diff
@@ -151,10 +152,13 @@ public void onFailure(Exception e) {
         IndicesOptions indicesOptions = IndicesOptions.fromOptions(ignoreUnavailable, true, true, false, IndicesOptions.lenientExpandOpen());
         FutureActionListener<RestoreSnapshotResponse, Long> listener = new FutureActionListener<>(r -> 1L);
         resolveIndexNames(analysis.restoreTables(), ignoreUnavailable, transportActionProvider.transportGetSnapshotsAction(), analysis.repositoryName())
-            .whenComplete((List<String> indexNames, Throwable t) -> {
+            .whenComplete((ResolveIndicesAndTemplatesContext resolvedIndicesAndSnapshots, Throwable t) -> {
                 if (t == null) {
+                    List<String> indexNames = ImmutableList.copyOf(resolvedIndicesAndSnapshots.resolvedIndices);
```

Comment:
can we avoid copying the collection/set here? at least, create an string array, otherwise all would be copied again at RestoreSnapshoptRequest